ast/compile.go

```diff
@@ -883,6 +883,60 @@ func compileSchema(goSchema interface{}) (*gojsonschema.Schema, error) {
 	return schemasCompiled, nil
 }
 
+//Checks if a JSON schema contains a specific type
+func containsType(schema *gojsonschema.SubSchema, s string) bool {
+	return schema.Types.String() == s
+}
+
+//Merges JSON schemas
+func mergeSchemas(schemas []*gojsonschema.SubSchema) (*gojsonschema.SubSchema, error) {
+	var result = schemas[0]
+	if len(schemas) == 0 {
+		return result, nil
+	}
+
+	for i := 0; i < len(schemas); i++ {
+		if len(schemas[i].PropertiesChildren) > 0 {
+			if !schemas[i].Types.Contains("object") {
+				schemas[i].Types.Add("object")
+			}
+		} else if len(schemas[i].ItemsChildren) > 0 {
+			if !schemas[i].Types.Contains("array") {
+				schemas[i].Types.Add("array")
+			}
+		}
+	}
+
+	for i := 1; i < len(schemas); i++ {
+		if result.Types.Contains("object") && len(result.PropertiesChildren) > 0 && schemas[i].Types.Contains("object") && len(schemas[i].PropertiesChildren) > 0 {
+			for j := 0; j < len(schemas[i].PropertiesChildren); j++ {
+				result.PropertiesChildren = append(result.PropertiesChildren, schemas[i].PropertiesChildren[j])
+			}
+		} else if result.Types.Contains("array") && len(result.ItemsChildren) > 0 && schemas[i].Types.Contains("array") && len(schemas[i].ItemsChildren) > 0 {
+			for j := 0; j < len(schemas[i].ItemsChildren); j++ {
+				if result.ItemsChildren[j].Types.String() != schemas[i].ItemsChildren[j].Types.String() {
+					return nil, fmt.Errorf("unable to merge these schemas due to different types")
+				} else if !containsType(result.ItemsChildren[j], schemas[i].ItemsChildren[j].Types.String()) {
+					result.ItemsChildren = append(result.ItemsChildren, schemas[i].ItemsChildren[j])
+				}
+			}
+		} else if (result.Types.Contains("object") && schemas[i].Types.Contains("array")) || (result.Types.Contains("array") && schemas[i].Types.Contains("object")) {
+			return nil, fmt.Errorf("unable to merge these schemas")
+		} else if result.Types.Contains("string") && schemas[i].Types.Contains("string") {
+			return result, nil
```

Comment:
Handling of `string`, `integer` and `boolean` isn't quite right. You are returning `result` as soon as you get `schemas[i]` that matches result type....what about the rest of the schemas??? 
Try this:

```
`{
	"$schema": "http://json-schema.org/draft-04/schema#",
	"type": "string",
	"title": "The b schema",
	"description": "An explanation about the purpose of this instance.",
	"allOf": [{
			"type": "string",
		},
		{
			"type": "string",
		},
		{
			"type": "boolean",
		}
	]
}`
```

This returns a merged type of `string` when it should have been failing with cannot merge because only schemas[0] is compared with schemas[1].....schemas[2], schemas[3] and so on are totally ignored